iOS workshop/Screens/Home/View/HomeViewController.swift

```diff
@@ -0,0 +1,126 @@
+    var selectedIndexPath: IndexPath?
+    var meals : [Meal] = []
+    @IBOutlet weak var tableview: UITableView!
+    @IBOutlet weak var categoryCollectionView: UICollectionView!
+    let viewModel = HomeVIewModel()
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableview.delegate = self
+        tableview.dataSource = self
+        tableview.separatorStyle = .none
+        
+        categoryCollectionView.register(UINib(nibName: "FoodCategoriesCollectionViewCell", bundle: .main), forCellWithReuseIdentifier: "FoodCategoriesCollectionViewCell")
+        
+        tableview.register(UINib(nibName: "RecipeCell", bundle: .main), forCellReuseIdentifier: "RecipeCell")
+        
+        categoryCollectionView.delegate = self
+        categoryCollectionView.dataSource = self
+        
+        let layeout = UICollectionViewFlowLayout()
+        layeout.itemSize = CGSize(width: 72, height: 96)
+        layeout.scrollDirection = .horizontal
+        layeout.minimumInteritemSpacing = 0
+        layeout.minimumLineSpacing = 5
+        
+        let titleLabel = UILabel()
+          titleLabel.text = "Food Recipes"
+          titleLabel.textAlignment = .center
+          titleLabel.font = UIFont.boldSystemFont(ofSize: 24) // Customize the font if needed
+          titleLabel.sizeToFit()
+        navigationItem.titleView = titleLabel
+        
+        navigationItem.largeTitleDisplayMode = .never
+
+        
+        categoryCollectionView.collectionViewLayout = layeout
+        
+        viewModel.getData(foodTag: Categories.getCategories()[0].categorieKey)
+        
+        viewModel.cellDataSource.bind(){ [weak self] meal in
+            guard let meals = meal else {
+                return
+            }
+            self?.meals = meals.map{ $0 }
+            self?.tableview.reloadData()
+        }
+        // Do any additional setup after loading the view.
+    }
+
+
+    /*
+    // MARK: - Navigation
+
+    // In a storyboard-based application, you will often want to do a little preparation before navigation
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        // Get the new view controller using segue.destination.
+        // Pass the selected object to the new view controller.
+    }
+    */
+
+}
+extension HomeViewController:UICollectionViewDelegate,UICollectionViewDataSource{
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return Categories.getCategories().count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = categoryCollectionView.dequeueReusableCell(withReuseIdentifier: "FoodCategoriesCollectionViewCell", for: indexPath) as! FoodCategoriesCollectionViewCell
+        let categories = Categories.getCategories()
+        cell.categoriesImage.image  = UIImage(named: categories[indexPath.row].categorieImage)
```

Comment:
add cell configuration in the cell class and call here this configuration method